File: src/main/kotlin/com/ks/culinario/data/entity/RecipeEntity.kt
package com.ks.culinario.data.entity

import jakarta.persistence.*

@Entity
@Table(name = "recipes")
data class RecipeEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,

    val title: String = "",
    val description: String = "",

    val stepTitle: String = "",
    val stepDescription: String = "",

    val owner: String = "",
    val category: String = "",
    val created: String = "",

    val shoppingListId: Int? = null
)

File: dotnet/Culinario.Infrastructure/Persistence/Entities/RecipeEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Culinario.Infrastructure.Persistence.Entities;

[Table("Recipes")]
public class RecipeEntity
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int? Id { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    public string StepsJson { get; set; } = string.Empty;

    public string Owner { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string Created { get; set; } = string.Empty;

    public int? ShoppingListId { get; set; }
}
